### ISSUE TITLE
TORCH_DISTRIBUTED_DEBUG should print the expected CollectiveFingerPrint

### ISSUE THREAD

**Issue by pritamdamania87** [pytorch/pytorch#78022]
### 🚀 The feature, motivation and pitch

During collective mismatch detection, we print errors as follows:

```
RuntimeError: Detected mismatch between collectives on ranks. Rank 0 is running inconsistent collective: CollectiveFingerPrint(OpType=BROADCAST, TensorShape=[34112], TensorDtypes=Float, TensorDeviceTypes=TensorOptions(dtype=float (default), device=cuda, layout=Strided (default), requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
```

However, the problem here is that we don't actually print what is the expected CollectiveFingerPrint. Adding this information would make it much easier to debug issues.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @kwen2501

**Comment by rohan-varma** [comment]
Agreed this would be very helpful. Is there a particular use case where this came up? 

Currently, we work around this by looking at logs for each rank and finding this msg, then checking which collective is different across ranks. Quite manual and not scalable, but for smaller scale jobs it might be helpful?

**Comment by pritamdamania87** [comment]
> Agreed this would be very helpful. Is there a particular use case where this came up?

Yes it came up here: https://discuss.pytorch.org/t/detected-mismatch-between-collectives-on-ranks/151846 and I do think we should enhance this such that we don't have to look at all logs.

**Comment by rohan-varma** [comment]
cc @H-Huang who mentioned interest in working on TDD

**Comment by H-Huang** [comment]
Resolved with #79724 #79901
